book-1-orientation/chapters/QUEST.md

```diff
@@ -0,0 +1,287 @@
+            private int _correctAnswer;
+            private int _awesomenessChange;
+
+
+            // A constructor for the Challenge
+            // We can tell it's a constructor because it has the same name as the class 
+            //   and it doesn't specify a return type
+            // Note the constructor parameters and what is done with them
+            public Challenge(string text, int correctAnswer, int awesomenessChange)
+            {
+                _text = text;
+                _correctAnswer = correctAnswer;
+                _awesomenessChange = awesomenessChange;
+            }
+
+            // This method will take a Adventurer object and make that Adventurer perform the challenge
+            public void RunChallenge(Adventurer Adventurer)
+            {
+                Console.Write($"{_text}: ");
+                string answer = Console.ReadLine();
+
+                int numAnswer;
+                bool isNumber = int.TryParse(answer, out numAnswer);
+
+                Console.WriteLine();
+                if (isNumber && numAnswer == _correctAnswer)
+                {
+                    Console.WriteLine("Well Done!");
+
+                    // Note how we access a Adventurer object's property
+                    Adventurer.Awesomness += _awesomenessChange;
+                }
+                else
+                {
+                    Console.WriteLine("You have failed the challenge, there will be consequences.");
+                    Adventurer.Awesomness -= _awesomenessChange;
+                }
+
+                // Note how we call a Adventurer object's method
+                Console.WriteLine(Adventurer.GetAdventurerStatus());
+                Console.WriteLine();
+            }
+        }
+    }
+    ```
+
+1. Read over the code. Imagine what the code will do when it runs. Write some notes about what you think will happen.
+1. Run the program. Does it do what you expected?
+
+> **NOTE:** At the end of each phase you should Add, Commit and Push to GitHub.
+
+### Phase 2
+
+Our adventurer is currently named "Jack". Studies show that "Jack" is probably not the application user's name. Update the code to prompt the user for their name and pass that name to the `Adventurer` constructor when creating the new `Adventurer` object.
+
+### Phase 3
+
+This quest is so much fun that everyone is sure to want to do it more than once. Update the code to ask the user if they'd like to repeat the quest after the it has been completed. If the user says "yes", start the quest over. Otherwise, end the program.
+
+### Phase 4
+
+I don't know if you noticed, but our adventurer is naked. What happened to all that talk about a brightly colored robe and shiny hat?
+
+In this phase we'll add the robe.
+
+1. Create new file called `Robe.cs`.
+1. Inside the file define a `Robe` class.
+1. The class should have the following mutable properties.
+    * Colors - a list of strings to hold the colors of the robe
```

Comment:
Briefly thought you meant literally strings, as in thread. It's late. I'm easily confused